cmake/third_party/aws_sdk.cmake

```diff
@@ -1,44 +1,73 @@
+set(aws_sdk_include_location)
+
+if(IMPORT_S3_SDK_PACKAGE)
+    find_package(AWSSDK REQUIRED COMPONENTS s3-crt)
+    # Use the AWS provided variables to set the paths for the AWS targets.
+    set(s3_crt_lib_location ${AWSSDK_LIB_DIR}/libaws-cpp-sdk-s3-crt${CMAKE_SHARED_LIBRARY_SUFFIX})
+    set(aws_core_lib_location ${AWSSDK_LIB_DIR}/libaws-cpp-sdk-core${CMAKE_SHARED_LIBRARY_SUFFIX})
+    set(aws_sdk_include_location ${AWSSDK_INCLUDE_DIR})
+elseif(IMPORT_S3_SDK_EXTERNAL)
+    # Download and install the AWS CPP SDK into the build directory.
+    ExternalProject_Add(aws-sdk
+        PREFIX aws-sdk-cpp
+        GIT_REPOSITORY https://github.com/aws/aws-sdk-cpp.git
+        GIT_TAG 1.9.175
+        CMAKE_ARGS
+            -DBUILD_SHARED_LIBS=ON
+            -DBUILD_ONLY=s3-crt
+            -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/aws-sdk-cpp/install
+            -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+        BUILD_ALWAYS FALSE
+        INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/aws-sdk-cpp/install
+        BUILD_BYPRODUCTS
+            ${CMAKE_CURRENT_BINARY_DIR}/aws-sdk-cpp/install/${CMAKE_INSTALL_LIBDIR}/libaws-cpp-sdk-s3-crt${CMAKE_SHARED_LIBRARY_SUFFIX}
+            ${CMAKE_CURRENT_BINARY_DIR}/aws-sdk-cpp/install/${CMAKE_INSTALL_LIBDIR}/libaws-cpp-sdk-core${CMAKE_SHARED_LIBRARY_SUFFIX}
+        TEST_COMMAND ""
+        UPDATE_COMMAND ""
+    )
+    ExternalProject_Get_Property(aws-sdk INSTALL_DIR)
+    file(MAKE_DIRECTORY ${INSTALL_DIR}/${CMAKE_INSTALL_INCLUDEDIR})
+    # Set the path variables to be used for the AWS targets.
+    set(s3_crt_lib_location ${INSTALL_DIR}/${CMAKE_INSTALL_LIBDIR}/libaws-cpp-sdk-s3-crt${CMAKE_SHARED_LIBRARY_SUFFIX})
+    set(aws_core_lib_location ${INSTALL_DIR}/${CMAKE_INSTALL_LIBDIR}/libaws-cpp-sdk-core${CMAKE_SHARED_LIBRARY_SUFFIX})
+    set(aws_sdk_include_location ${INSTALL_DIR}/${CMAKE_INSTALL_INCLUDEDIR})
+endif()
 
 add_library(aws-sdk::core SHARED IMPORTED)
 add_library(aws-sdk::s3-crt SHARED IMPORTED)
 add_library(aws-sdk::crt SHARED IMPORTED)
 
 # Small workaround to declare the include directory under INTERFACE_INCLUDE_DIRECTORIES during the configuration phase.
-file(MAKE_DIRECTORY ${INSTALL_DIR}/${CMAKE_INSTALL_INCLUDEDIR})
 set_target_properties(aws-sdk::core PROPERTIES
-    IMPORTED_LOCATION ${INSTALL_DIR}/${CMAKE_INSTALL_LIBDIR}/libaws-cpp-sdk-core${CMAKE_SHARED_LIBRARY_SUFFIX}
-    INTERFACE_INCLUDE_DIRECTORIES ${INSTALL_DIR}/${CMAKE_INSTALL_INCLUDEDIR}                                      
+    IMPORTED_LOCATION ${aws_core_lib_location}
+    INTERFACE_INCLUDE_DIRECTORIES ${aws_sdk_include_location}
 )
 set_target_properties(aws-sdk::s3-crt PROPERTIES
-    IMPORTED_LOCATION ${INSTALL_DIR}/${CMAKE_INSTALL_LIBDIR}/libaws-cpp-sdk-s3-crt${CMAKE_SHARED_LIBRARY_SUFFIX}
-    INTERFACE_INCLUDE_DIRECTORIES ${INSTALL_DIR}/${CMAKE_INSTALL_INCLUDEDIR}
+    IMPORTED_LOCATION ${s3_crt_lib_location}
+    INTERFACE_INCLUDE_DIRECTORIES ${aws_sdk_include_location}
 )
 
 add_dependencies(aws-sdk::core aws-sdk)
```

Comment:
This should only happen if `IMPORT_S3_SDK_EXTERNAL` is set, since `aws-sdk` is the target associated with `ExternalProject_Add`. I'd just add an if statement around these two lines `if(IMPORT_S3_SDK_EXTERNAL)`